src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/SourceRepositoryExtensions.cs

```diff
@@ -114,6 +115,7 @@ public static Task<SearchResult<IPackageSearchMetadata>> SearchAsync(this Source
                     .FirstOrDefault(p => p.Identity.Version == identity.Version)
                     ?? PackageSearchMetadataBuilder.FromIdentity(identity).Build();
 
+                cancellationToken.ThrowIfCancellationRequested();
                 return packageMetadata.WithVersions(ToVersionInfo(packages, includePrerelease));
```

Comment:
I don't imagine that `packageMetadata.WithVersions(ToVersionInfo(packages, includePrerelease))` is very CPU intensive, so I don't think we should throw when the method is effectively finished.